wagtail_shortcode/static_src/main.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+
+const { RichUtils } = window.DraftJS;
+const { TooltipEntity } = window.draftail;
+const { Icon } = window.wagtail.components;
+
+type WagtailShortcodeConfig = {
+  helpText: string;
+};
+
+const getShortcodeConfig = (): WagtailShortcodeConfig => {
+  // eslint-disable-next-line no-undef
+  const configurationElement = document.querySelector<HTMLScriptElement>(
+    '#wagtail-shortcode-config',
+  );
+  if (!configurationElement || !configurationElement.textContent) {
+    throw new Error('No wagtail-shortcode configuration found.');
+  }
+
+  try {
+    return JSON.parse(configurationElement.textContent);
+  } catch (err) {
+    throw new SyntaxError(
+      `Error parsing wagtail-shortcode configuration: ${err.message}`,
+    );
+  }
+};
+
+/** Responsible for creating new entity instances in the editor. */
+class ShortcodeSource extends window.React.Component {
+  componentDidMount() {
+    const { editorState, entityType, entity, entityKey, onComplete }: any =
+      this.props;
+
+    const content = editorState.getCurrentContent();
+
+    let defaultValue;
+
+    if (entity && entityKey) {
+      // If an existing entity get the shortcode value to provide as default
+      const { shortcode } = entity.getData();
+      defaultValue = shortcode;
+    }
+
+    const { helpText } = getShortcodeConfig();
+
+    // eslint-disable-next-line no-alert
+    const shortcodeValue = window.prompt(helpText, defaultValue);
```

Comment:
**Note:**
As mentioned on slack, there are obvious limitations to basic text formating, if needed I'll open up a seperate PR with draft code for adding a model.